http-server/src/test/java/org/triplea/server/access/RoleAuthorizerTest.java

```diff
@@ -18,45 +17,60 @@
+        .thenReturn(UserRole.ADMIN)
+        .thenReturn(UserRole.MODERATOR)
+        .thenReturn(UserRole.PLAYER)
+        .thenReturn(UserRole.ANONYMOUS);
+
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.ADMIN), is(true));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.ADMIN), is(false));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.ADMIN), is(false));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.ADMIN), is(false));
   }
 
-  @Nested
-  final class AuthenticateModerator {
-    @Test
-    void authenticateModerator() {
-      when(authenticatedUser.isModerator()).thenReturn(true);
-
-      final boolean result = roleAuthorizer.authorize(authenticatedUser, UserRole.MODERATOR);
-
-      assertThat(result, is(true));
-    }
-
-    @Test
-    void authenticateModeratorNegativeCase() {
-      when(authenticatedUser.isModerator()).thenReturn(false);
+  @Test
+  void authenticateModerator() {
+    when(authenticatedUser.getUserRole())
+        .thenReturn(UserRole.ADMIN)
+        .thenReturn(UserRole.MODERATOR)
+        .thenReturn(UserRole.PLAYER)
+        .thenReturn(UserRole.ANONYMOUS);
+
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.MODERATOR), is(true));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.MODERATOR), is(true));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.MODERATOR), is(false));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.MODERATOR), is(false));
+  }
 
-      final boolean result = roleAuthorizer.authorize(authenticatedUser, UserRole.MODERATOR);
+  @Test
+  void authenticatePlayer() {
+    when(authenticatedUser.getUserRole())
+        .thenReturn(UserRole.ADMIN)
+        .thenReturn(UserRole.MODERATOR)
+        .thenReturn(UserRole.PLAYER)
+        .thenReturn(UserRole.ANONYMOUS);
+
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.PLAYER), is(true));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.PLAYER), is(true));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.PLAYER), is(true));
+    // on fourth iteration, role is anonymous, not authorized
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.PLAYER), is(false));
+  }
 
-      assertThat(result, is(false));
-    }
+  @Test
+  void authenticateAnonymous() {
+    when(authenticatedUser.getUserRole())
+        .thenReturn(UserRole.ADMIN)
+        .thenReturn(UserRole.MODERATOR)
+        .thenReturn(UserRole.PLAYER)
+        .thenReturn(UserRole.ANONYMOUS);
+
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.ANONYMOUS), is(true));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.ANONYMOUS), is(true));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.ANONYMOUS), is(true));
+    assertThat(roleAuthorizer.authorize(authenticatedUser, UserRole.ANONYMOUS), is(true));
```

Comment:
These unit tests seem backwards to me? Shouldn't an admin return true for all and anonymous only return true for anonymous role?

Reply:
This unit test is basically saying:

Can an admin authenticate as anonymous? yes
Can a moderator authenticate as anonymous ? yes
etc..

It's not quite the API I would have chosen for this, but there is a constraint to work within the framework. Creating a delegate object to handle authorization I think would add too much indirection.

Reply:
Ah, I see now. I read the test methods in the opposite way where I thought it was saying let's check admin user against the 4 roles rather than let's check each user type against admin role. I think maybe updating the unit test names to "authenticateAsAdmin" instead of "authenticateAdmin" would make that more clear.

Reply:
Good suggestion, I'll take a look at the test method names and see if they can be improved.